### PR TITLE
Build MA57 even without get_factors.patch

### PR DESCRIPTION
**Pull request by abelsiqueira**
The large changes are actually just moving code from one file to another.

### CODE REVIEW

File: src/hsl_ma57.jl [hunk 1]
```diff
@@ -473,347 +473,3 @@ function ma57_least_squares(A :: SparseMatrixCSC{T,Ti}, b :: Array{T}) where {T
...
-    end
-    ka = ka + nrows * (ncols - nrows)
-    kw = kw + ncols + 2
-  end
-end
-
-
-## additional docstrings -------------------------------------------------------
-"""
-# Factorize `Ma57` object.
```
Comment (dpo) [inline src/hsl_ma57.jl]: These docstrings are not related to the patch. Did you also mean to move them?

Comment (abelsiqueira) [reply src/hsl_ma57.jl]: No, I only saw the last lines. I fixed now. Only the appropriate docstrings are moved.

File: deps/build_hsl_ma57.jl [hunk 2]
```diff
@@ -8,7 +8,9 @@ end
 run(`tar -zxf $hsl_ma57_archive -C $builddir`)
 cd("$builddir/hsl_ma57-$hsl_ma57_version")
 patchfile = joinpath(@__DIR__, "downloads", "get_factors.patch")
```
Comment (dpo) [inline deps/build_hsl_ma57.jl]: Should this be `hsl_ma57_patch` now?

Comment (abelsiqueira) [reply deps/build_hsl_ma57.jl]: Fixed.

**Comment by dpo** [comment]
Works for me both with and without the patch! Thank you!
